cmd/goplugin-helm/resource/release.go

```diff
@@ -0,0 +1,177 @@
+package resource
+
+import (
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"strconv"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/lyraproj/servicesdk/serviceapi"
+)
+
+// ReleaseList is used for unmarhsalling the helm ls json output
+type ReleaseList struct {
+	Releases []Release
+}
+
+// Release is a helm release that can be created, updated or deleted
+type Release struct {
+	Name      string
+	Chart     string
+	Overrides *map[string]string
+	Namespace *string
+	Status    *string
+	Output    *string
+	Timeout   *int
+}
+
+// ReleaseHandler is used to perform CRUD operations on a Release resource
+type ReleaseHandler struct{}
+
+// Create a new Release in helm
+func (h *ReleaseHandler) Create(desiredState *Release) (*Release, string, error) {
+	hclog.Default().Debug("Creating Release", "desiredState", desiredState)
+	externalID := extID(*desiredState)
+	release, err := h.upsertRelease(externalID, desiredState)
+	return release, externalID, err
+}
+
+// Read an existing Release
+func (*ReleaseHandler) Read(externalID string) (*Release, error) {
+	log := hclog.Default()
+	log.Debug("Reading Release", "externalID", externalID)
+
+	//ensure helm is initialised
+	_, err := runCmd("", "helm", "init")
+	if err != nil {
+		return nil, err
+	}
+	s, err := runCmd("", "helm", "ls", "-a", "--output", "json")
+	if err != nil {
+		return nil, err
+	}
+	//an empty response means no releases
+	if len(s) == 0 {
+		return nil, serviceapi.NotFound("release", externalID)
+	}
+	var releaseList ReleaseList
+	b := []byte(s)
+	err = json.Unmarshal(b, &releaseList)
+	if err != nil {
+		panic(fmt.Errorf("unable to unmarshal, err is %v and output is %s", err, b))
```

Comment:
Why use panic here but `return nil, err` on line 52?

Reply:
agreed, left in by mistake.